lib/module.js

```diff
@@ -285,12 +276,11 @@ Module._resolveLookupPaths = function(request, parent) {
     return [request, []];
   }
 
-  var reqLen = request.length;
   // Check for relative path
-  if (reqLen < 2 ||
-      request.charCodeAt(0) !== 46/*.*/ ||
-      (request.charCodeAt(1) !== 46/*.*/ &&
-       request.charCodeAt(1) !== 47/*/*/)) {
+  if (request.charCodeAt(0) !== 46/*.*/ || (
+      request.charCodeAt(1) !== 46/*.*/ &&
+      request.charCodeAt(1) !== 47/*/*/ &&
+      request.length > 1)) {
```

Comment:
You'll want to add request length checking *before* using `charCodeAt(n)` to make sure you don't cause a deopt due to an out-of-bounds access.